test/success.test.js

```diff
@@ -2269,6 +2288,7 @@ test('Skip closing issues if "failComment" is "false"', async (t) => {
     .sandbox()
     .getOnce(`https://api.github.local/repos/${owner}/${repo}`, {
       full_name: `${owner}/${repo}`,
+      git_url: `https://api.github.local/${owner}/${repo}.git`,
```

Comment:
Let's write a test case where the `git_url` is 

- without `https`
- with `ssh`
- ...and any other format of `git_url` possible.

Reply:
okay, I think I should put those tests in `verify.test.js` though, this comment is in `success.test.js` but the success step doesn't do anything with `git_url`

Reply:
@babblebey you know, I think we should go back to `clone_url`.  There are two issues I see with `git_url`:
- It's always `git://` protocol, but GH docs say you can only push to HTTPS or SSH URLs.  (the git protocol has no authentication).  Not that it really makes a difference, but `clone_url` is HTTPS, and there's always `ssh_url`.
- The example for `git_url` in GH docs is `git:github.com/octocat/Hello-World.git`, which isn't a valid URL.  The api seems to always return `git://`, but this just makes me feel more wary of that field...

Reply:
> you know, I think we should go back to clone_url, or ssh_url. 

Let's do it @jedwards1211 ✊🏾

We must only ensure we have a test covering all the possible formats of the `clone_url`... I found this, maybe it can help 🙂

https://git-scm.com/docs/git-clone/en#_git_urls

Reply:
> I'm pretty sure clone_url is always HTTPS (it doesn't seem like any of these fields can vary in their format)

Though I think there could be more format eh, varying from `http`, `https`, to `ssh`, `git` (itself) and even `ftp`....referring to the git docs https://git-scm.com/docs/git-clone/en#_git_urls

....and considering your comment https://github.com/semantic-release/github/issues/885#issuecomment-2251402572 where I see an `ssh` looking kind of `clone_url`

🤔

Reply:
Oh well yes, git itself accepts many different URL formats, but GitHub is in control of what clone_url it returns, so I think it always uses the same format.  But I'll double check if there's any way for users to configure the clone_url format.

The ssh-like URLs in my comment on #885 didn't come from the GitHub API, but rather, when I tried to `git remote set-url` with a normal HTTPS URL, for some reason git stored a more SSH style URL instead.  So if you want me to test the weird URL formats stored in git remotes then I should set `repositoryUrl` to the weird formats in the tests, rather than `clone_url` or `git_url`.

Reply:
> So if you want me to test the weird URL formats stored in git remotes then I should set `repositoryUrl` to the weird formats, rather than `clone_url` or `git_url`.

Oh no 😃, the `repositoryUrl` cannot be the weird format, this is because the `repositoryUrl` is the already parsed url gotten of the project from either its `package.json` or semantic-realease config.

It is the value we're trying to verify to confirm whether its correct/updated by fetching repo data... main reason for this PR 😉

Reply:
What I mean is the sematic-release docs say for `repositoryUrl`:
> Any valid git url format is supported

So there are a lot of options for what users could configure it to.

Have you seen an example of the GitHub API returning a `clone_url` that isn't in HTTPS format?

Reply:
Okay I switched back to `clone_url` and make the tests go through a matrix of possible combinations for `clone_url` and `repositoryUrl` formats.

Reply:
> make the tests go through a matrix of possible combinations for `clone_url` and `repositoryUrl` formats.

Haha 😆 

Well, there's literally a matrix possible format of `repositoryUrl` any way, considering that depends on the format configured for `semantic-release` per project yea... and maybe same for `clone_url`... But the thing that matters most in there is the `owner/repo` 😉

Reply:
Yup.  Let me know if I need to do anything else on this